File: src/main/java/de/hsrm/mi/abschlussarbeit/customerRequestProcessingServer/capacity/CalculatedCapacitiesOfMatchDto.java
package de.hsrm.mi.abschlussarbeit.customerRequestProcessingServer.capacity;

import de.hsrm.mi.abschlussarbeit.customerRequestProcessingServer.employee.EmployeeDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CalculatedCapacitiesOfMatchDto {
    private EmployeeDto employee;
    private Long expertisePoints;
    private Boolean canCompleteTaskEarliest;
    private List<CalculatedCapacityCalendarEntryDto> calculatedCalendarCapacities;
}

File: frontend/Workstation/src/documentTypes/dtds/CalculatedCapacitiesOfMatchDto.ts
import type {EmployeeDtd} from "@/documentTypes/dtds/EmployeeDtd.ts";
import type {
  CalculatedCapacityCalendarEntryDtd
} from "@/documentTypes/dtds/CalculatedCapacityCalendarEntryDtd .ts";
import type {CalendarDtd} from "@/documentTypes/dtds/CalendarDtd.ts";

export interface CalculatedCapacitiesOfMatchDto {
  employee: EmployeeDtd
  expertisePoints: number
  canCompleteTaskEarliest: boolean
  calculatedCalendarCapacities: CalculatedCapacityCalendarEntryDtd[]
  calendar?: CalendarDtd
}
